be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ApiControllers/SearchSuggestionsApiController.cs

```diff
@@ -13,23 +16,42 @@ namespace Buyer.Web.Areas.Products.ApiControllers
     [Area("Products")]
     public class SearchSuggestionsApiController : BaseApiController
     {
-        private readonly IProductsService productsService;
+        private readonly IProductsService _productsService;
+        private readonly IInventoryService _inventoryService;
 
-        public SearchSuggestionsApiController(IProductsService productsService)
+        public SearchSuggestionsApiController(
+            IProductsService productsService,
+            IInventoryService inventoryService)
         {
-            this.productsService = productsService;
+            _productsService = productsService;
+            _inventoryService = inventoryService;
         }
 
         [HttpGet]
-        public async Task<IActionResult> Get(string searchTerm, int size)
+        public async Task<IActionResult> Get(string searchTerm, int size, string searchArea)
         {
-            var suggestions = await this.productsService.GetProductSuggestionsAsync(
-                searchTerm,
-                size,
-                CultureInfo.CurrentUICulture.Name,
-                await HttpContext.GetTokenAsync(ApiExtensionsConstants.TokenName));
+            IEnumerable<ProductSuggestionViewModel> suggestions;
 
-            return this.StatusCode((int)HttpStatusCode.OK, suggestions);
+            if (searchArea == SearchConstants.SearchArea.StockLevel)
```

Comment:
**suggestion:** it would be good to add here the strategy design pattern, please ask chat gpt how to do it